lib/parse.js

```diff
@@ -36,13 +40,20 @@ function parseNonShell(parsed) {
     if (needsShell) {
         // Escape command & arguments
         applyQuotes = (parsed.command !== 'echo');  // Do not quote arguments for the special "echo" command
-        parsed.command = escapeCommand(parsed.command);
```

Comment:
Keeping everything on the `parsed` object is more confusing to a new reader.  Doing so would mean we are storing a temporary value onto the object with no intention of keeping that value, since it is overwritten on line 57.  By using a variable, we're indicating that this is a temporary value used in computation.  `command` would ideally be declared `const` to make the code's intent even clearer.